source/dataman/MdtmMan.cpp

```diff
@@ -0,0 +1,250 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ *
+ * MdtmMan.cpp
+ *
+ *  Created on: Apr 20, 2017
+ *      Author: Jason Wang
+ */
+
+#include "MdtmMan.h"
+
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "zmq.h"
+
+MdtmMan::~MdtmMan()
+{
+    if (zmq_ipc_req)
+        zmq_close(zmq_ipc_req);
+}
+
+int MdtmMan::init(json p_jmsg)
+{
+
+    StreamMan::init(p_jmsg);
+
+    if (p_jmsg["pipe_prefix"] == nullptr)
+    {
+        pipe_desc["pipe_prefix"] = "/tmp/MdtmManPipes/";
+    }
+    else
+    {
+        pipe_desc["pipe_prefix"] = p_jmsg["pipe_prefix"];
+    }
+
+    pipe_desc["operation"] = "init";
+    pipe_desc["mode"] = m_stream_mode;
+
+    std::string pipename_prefix = "MdtmManPipe";
+    for (int i = 0; i < m_num_channels; i++)
+    {
+        std::stringstream pipename;
+        pipename << pipename_prefix << i;
+        if (i == 0)
+        {
+            pipe_desc["pipe_names"] = {pipename.str()};
+            pipe_desc["loss_tolerance"] = {m_tolerance[i]};
+            pipe_desc["priority"] = {m_priority[i]};
+        }
+        else
+        {
+            pipe_desc["pipe_names"].insert(pipe_desc["pipe_names"].end(),
+                                           pipename.str());
+            pipe_desc["loss_tolerance"].insert(
+                pipe_desc["loss_tolerance"].end(), m_tolerance[i]);
+            pipe_desc["priority"].insert(pipe_desc["priority"].end(),
+                                         m_priority[i]);
+        }
+    }
+
+    // ZMQ_DataMan_MDTM
+    if (m_stream_mode == "sender")
+    {
+        zmq_ipc_req = zmq_socket(zmq_context, ZMQ_REQ);
+        zmq_connect(zmq_ipc_req, "ipc:///tmp/ADIOS_MDTM_pipe");
+        char buffer_return[10];
+        zmq_send(zmq_ipc_req, pipe_desc.dump().c_str(),
+                 pipe_desc.dump().length(), 0);
+        zmq_recv(zmq_ipc_req, buffer_return, 10, 0);
```

Comment:
Use `sizeof(buffer_return)` here instead of `10` in case you ever want to change buffer_return's size.